src/conf_mode/load-balancing-haproxy.py

```diff
@@ -0,0 +1,125 @@
+#!/usr/bin/env python3
+#
+# Copyright (C) 2023 VyOS maintainers and contributors
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 2 or later as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import os
+
+from sys import exit
+from shutil import rmtree
+
+from vyos.base import Warning
+from vyos.config import Config
+from vyos.configdict import dict_merge
+from vyos.util import call
+from vyos.template import render
+from vyos.xml import defaults
+from vyos import ConfigError
+from vyos import airbag
+airbag.enable()
+
+load_balancing_dir = '/run/haproxy'
+load_balancing_conf_file = f'{load_balancing_dir}/haproxy.cfg'
+systemd_service = 'haproxy.service'
+systemd_override = r'/run/systemd/system/haproxy.service.d/10-override.conf'
+
+
+def get_config(config=None):
+    if config:
+        conf = config
+    else:
+        conf = Config()
+
+    base = ['load-balancing', 'reverse-proxy']
+    lb = conf.get_config_dict(base,
+                              get_first_key=True,
+                              key_mangling=('-', '_'),
+                              no_tag_node_value_mangle=True)
+
+    # We have gathered the dict representation of the CLI, but there are default
+    # options which we need to update into the dictionary retrived.
+    default_values = defaults(base)
+    if 'backend' in default_values:
+        del default_values['backend']
+    lb = dict_merge(default_values, lb)
+
+    if 'backend' in lb:
+        for backend in lb['backend']:
+            default_balues_backend = defaults(base + ['backend'])
+            lb['backend'][backend] = dict_merge(default_balues_backend, lb['backend'][backend])
+
+    return lb
+
+
+def verify(lb):
+    if not lb:
+        return None
+
+    if 'backend' not in lb or 'server' not in lb:
+        raise ConfigError(f'"server" and "backend" must be configured!')
+
+    for front, front_config in lb['server'].items():
+        if 'port' not in front_config:
+            raise ConfigError(f'"{front} server port" must be configured!')
+
```

Comment:
Should we check for bind ports already used?

Reply:
Done!